activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -180,7 +180,7 @@ def prepare_all
         each_current_configuration(env) do |db_config|
           database_initialized = initialize_database(db_config)
 
-          seed = true if database_initialized
+          seed = true if database_initialized && !db_config.framework?
```

Comment:
I think the problem is that this `seed` is a regular boolean when it should not. It isn't taking in consideration that the application can have multiple database. The other problem is that seeds are global and don't specify to which database those seeds are for.

I really think it is a bad idea to run a task that can run the seeds in production. Why are we doing that? Seeds are mostly for development.